Week3/Homework/task3_morejs.js

```diff
@@ -0,0 +1,260 @@
+        }
+    }
+    console.log(advertisementText);
+}
+
+advertiseWithAnotherLoop();
+
+// 11. What if you add one more vehicle to the list, can you have that 
+//     added to the advertisement without changing the code for question 10?
+
+vehicles.push("plane", "boat", "train");
+advertiseWithJoin();
+advertiseWithLoop();
+advertiseWithAnotherLoop();
+
+// 12. Create an empty object.
+
+const emptyObj = {};
+
+// 13. Create an object that contains the teachers that you 
+//     have had so far for the different modules.
+
+let hackYourFuture = {teachers: ["Philipp", "Rob", "Hardit", "Unmesh", "Sander", "Marciano"]};
+console.log(hackYourFuture);
+
+// 14. Add a property to the object you just created that 
+//     contains the languages that they have taught you.
+
+hackYourFuture.languages = ["html", "css", "Git", "javascript"];
+console.log(hackYourFuture);
+
+// 15. Write some code to test two arrays for equality using == and ===. 
+// Test the following:
+
+// let x = [1, 2, 3];
+// let y = [1, 2, 3];
+// let z = y;
+// What do you think will happen with x == y, x === y and z == y and z == x? Prove it!
+
+// Don't cheat! Seriously - try it first.
+
+// Check out this Fiddle. You need to open your browser’s Developer 
+// Tools to see the console output. Press the Run button in the upper right 
+// corner to run the code.
+
+// More insights from this Stack Overflow question.
+
+let x = [1, 2, 3];
+let y = [1, 2, 3];
+let z = y;
+
+console.log("x == y, x === y, z == y and z == x seems like true conditions. Let's try!");
+console.log("x == y : " + (x == y));
+console.log("x === y : " + (x === y));
+console.log("z == y : " + (z == y));
+console.log("z == x : " + (z == x));
+
+console.log("So there seems to be a special situation for arrays (or objects).");
+console.log("Even if two seperately created arrays have identical elements,"); 
+console.log("type, length, order, etc, they are still not the same for JS.");
+console.log("It is about the reference thing that objects have.");
+
+// 16. Take a look at the following code:
+
+// let o1 = { foo: "bar" };
+// let o2 = { foo: "bar" };
+// let o3 = o2;
+// Show that changing o2 changes o3 (or not) and changing o1 changes o3(or not).
+
+// Does the order that you assign (o3 = o2 or o2 = o3) matter?
+
+let o1 = { foo: "bar" };
+let o2 = { foo: "bar" };
+let o3 = o2;
+
+console.log(o3); // same as o2
+console.log("o3 === o2 : " + (o3 === o2)); // true
+console.log("o2 === o3 : " + (o2 === o3)); // true
+o2 = { boo : "far" };
+console.log(o2); // new o2
+console.log(o3); // does not change
+console.log("o2 === o3 : " + (o2 === o3)); //false
+
+o2 = o3;
+console.log(o2); // {foo: "bar"}
+console.log("o3 === o2 : " + (o3 === o2)); // true
+o2.boo = "far";
+console.log(o2); // {foo: "bar", boo: "far"}
+console.log(o3); // {foo: "bar", boo: "far"} -- o3 changes but why now? is it because of order?
+console.log("o3 === o2 : " + (o3 === o2)); // true
+o2 = { foo: "bar" };
```

Comment:
And then here you `re-assign` `o2`, which breaks the previous `link` of `o2 = o3`, so then if you would change `o3`, `o2` would not change with it.